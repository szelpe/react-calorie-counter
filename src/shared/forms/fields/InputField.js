import React from 'react';

export default function InputField(props) {
    return (
        <div className="form-group">
            <label htmlFor={props.id}>{props.labelText}</label>
            <input type={props.type}
                   value={props.value}
                   id={props.id}
                   name={props.id}
                   onChange={handleChange}
                   onBlur={handleBlur}
                   className="form-control"
            />
            <span className="text-danger">{props.error}</span>
        </div>
    );

    function handleChange(event) {
        props.onChange({
            name: event.target.name,
            value: event.target.value
        });
    }

    function handleBlur(event) {
        props.onBlur({
            name: event.target.name,
            value: event.target.value
        });
    }
}

InputField.defaultProps = {
    type: 'text'
};