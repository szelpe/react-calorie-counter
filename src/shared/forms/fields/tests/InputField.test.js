import React from 'react'
import { shallow, mount } from 'enzyme'

import InputField from "../InputField";
import renderer from 'react-test-renderer'

test('snapshot test', () => {
    let wrapper = renderer.create(<InputField />).toJSON();
    expect(wrapper).toMatchSnapshot();
});
