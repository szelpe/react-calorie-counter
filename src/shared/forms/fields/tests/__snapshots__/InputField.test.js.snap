// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="form-group"
>
  <label>
    test
  </label>
  <input
    class="form-control"
    type="text"
    value="test value"
  />
  <span
    class="text-danger"
  />
</div>
`;
